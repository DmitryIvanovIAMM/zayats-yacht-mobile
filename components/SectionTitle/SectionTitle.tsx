import { secondary } from "@/constants/Colors"; // Adjust the import path as necessary
import React from "react";
import { StyleProp, StyleSheet, Text, TextStyle, View } from "react-native";

type SectionTitleProps = {
  children: React.ReactNode;
  style?: StyleProp<TextStyle>;
};

const SectionTitle: React.FC<SectionTitleProps> = ({ children, style }) => (
  <View>
    <Text style={[styles.title, style]}>{children}</Text>
  </View>
);

const styles = StyleSheet.create({
  title: {
    fontSize: 24, // h2 equivalent
    fontWeight: "bold",
    color: secondary.dark, // Use the color constant
    textAlign: "center",
    marginVertical: 16,
  },
});

export default SectionTitle;
