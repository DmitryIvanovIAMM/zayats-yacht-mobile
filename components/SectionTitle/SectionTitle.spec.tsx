import { render } from "@testing-library/react-native";
import * as React from "react";
import { StyleSheet } from "react-native";
import SectionTitle from "./SectionTitle";

describe("SectionTitle", () => {
  it("matches snapshot", () => {
    const { toJSON } = render(<SectionTitle>Snapshot Title</SectionTitle>);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders children text", () => {
    const { getByText } = render(<SectionTitle>Testimonials</SectionTitle>);
    expect(getByText("Testimonials")).toBeTruthy();
  });

  it("applies custom style", () => {
    const { getByText } = render(
      <SectionTitle style={{ color: "red" }}>Styled</SectionTitle>
    );
    const text = getByText("Styled");
    const flattened = StyleSheet.flatten(text.props.style);
    expect(flattened.color).toBe("red");
  });
});
