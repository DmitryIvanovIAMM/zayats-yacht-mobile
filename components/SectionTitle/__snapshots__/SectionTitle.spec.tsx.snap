// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitle matches snapshot 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "width": "80%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#006666",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginVertical": 16,
          "textAlign": "center",
          "textTransform": "uppercase",
        },
        undefined,
      ]
    }
  >
    Snapshot Title
  </Text>
</View>
`;
